docs/guides/repository_v5_to_v6_migration.md

```diff
@@ -0,0 +1,49 @@
+---
+page_title: "Migrate repository resources from V5 to V6 provider"
+---
+
+# Migrate repository resources from V5 to V6 provider
+
+The migration strategy we recommend consists of two stages:
+1. In provider v5, migrate repository resources from 'generic' resources (`artifactory_local_repository`, `artifactory_remote_repository`, `artifactory_virtual_repository`) to package type specific resources (e.g. `artifactory_local_npm_repository`, `artifactory_remote_npm_repository`, `artifactory_virtual_npm_repository`).
+2. Upgrade Artifactory provider to v6.x
+
+## Migrate resources
+
+~>Before performing any major changes to Terraform configuration, it's highly recommended to backup your Terraform state.
+
+The major steps to migrate resources to new resource type are:
+1. Define new resources using package type specific resources in your configuration to match your infrastructure. This can be as straightforward as copying the attributes from existing resources to new resources and removing the package_type attribute.
+2. Import your repositories into Terraform state for these new resources
+  ```sh
+  $ terraform import artifactory_local_npm_repository.my-npm-local my-npm-local
+  ```
+3. Remove the original resource from Terraform state
+  ```sh
+  $ terraform state rm artifactory_local_repository.my-npm-local
+  ```
+
+## Upgrade provider to v6
+
+While it may be tempting to jump straight to the latest version (v6.21.3), there have been numerous new features and bug fixes that may make this large jump challenging to tackle. For this reason, we suggest picking an earlier version of v6 as the initial upgrade target. This should enable a smoother process and allow you to gain trust and confidence before upgrading to newer versions. Use `CHANGELOG.md` to guide you in this process.
```

Comment:
Is it not possible to give a recommended 6.x.x version to migrate too?

Or maybe a version that has the qualification, "We've found success migrating to version 6.x.x, but your results may vary. Use `Changelog.md` to help you find the right version for you`

I understand we don't want to give a version that then doesn't work for the customer causing backlash against us, but if I was a developer and there was no recommended version to migrate to I'd be a little irked.

Reply:
It depends on what the customer comfort level is, with respect to the new feature they want to take on. There's no 'right' answer so the customer has to make that determination theirselves using the CHANGELOG as guide.